content/blog/2023/09/03/2023-09-03-jsr305-probe-blog.adoc

```diff
@@ -0,0 +1,56 @@
+---
+layout: post
+title: Detecting deprecated JSR-305 imports Probe
+tags:
+- gsoc
+- gsoc2023
+- healthscore
+- probes
+- plugin
+authors:
+- jagruti
+opengraph:
+  image: /images/gsoc/2023/jagruti/jsr-305-probe-image.png
+sig:
+- gsoc
+---
+
+== Introduction
+
+This blog post describes the latest addition to the plugin health scoring system developed as part of link:../../../../projects/gsoc/2023/projects/add-probes-to-plugin-health-score.adoc[my Google summer of Code Project]: a probe that detects usages of the `JSR-305` framework in plugins.
+
+The link:https://jcp.org/en/jsr/detail?id=305[`JSR-305` framework] provides annotations to detect software defects.
+Its executive committee link:https://jcp.org/en/jsr/detail?id=305[voted it as dormant since May 2012].
+It is important to stop using the framework for the following reasons:
+
+- It is no longer maintained.
+- The annotations have questionable license.
+- The annotations are in the reserved javax namespace and there is no public release of the spec.
+Since it is not actively maintained, there is little chance this would be published.
+
+For more insight, refer to the link:https://groups.google.com/g/jenkinsci-dev/c/uE1wwtVi1W0/m/gLxdEJmlBQAJ[Jenkins developer mailing list conversation].
+
+The `@Nonnull` and `@CheckForNull` annotations, which are part of JSR-305, have been link:https://www.jenkins.io/doc/developer/tutorial-improve/replace-jsr-305-annotations/[deprecated since 2016].
```

Comment:
```suggestion
The `@Nonnull` and `@CheckForNull` annotations, which are part of JSR-305, have been link:/doc/developer/tutorial-improve/replace-jsr-305-annotations/[deprecated since 2016].
```
Suggestion to remove the extraneous part of this link